### PR TITLE
Extract `devToken()` outside of `ChatClient`

### PR DESCRIPTION
**Pull request by JcMinarro**
### 🎯 Goal
Extract `devToken()` outside of `ChatClient`. Move the implementation to `TokenUtils` and test it properly

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch
- ~[ ] PR is linked to the GitHub issue it resolves~

#### Code & documentation
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### CODE REVIEW

**Review by skydoves — APPROVED** [review]
Looks good to me :)

### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [x] Compose sample runs & works
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt [hunk 1]
```diff
@@ -2296,14 +2294,7 @@ internal constructor(
...
 
-    public fun devToken(userId: String): String {
-        require(userId.isNotEmpty()) { "User id must not be empty" }
-        val header = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9" //  {"alg": "HS256", "typ": "JWT"}
-        val devSignature = "devtoken"
-        val payload: String =
-            Base64.encodeToString("{\"user_id\":\"$userId\"}".toByteArray(StandardCharsets.UTF_8), Base64.NO_WRAP)
-        return "$header.$payload.$devSignature"
-    }
+    public fun devToken(userId: String): String = tokenUtils.devToken(userId)
```
Comment (adasiewiczr) [inline stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt:2297]: Would be nice to add KDoc here

**Comment by kanat** [comment]
Hey @JcMinarro,

Shouldn't we go with the process of deprecating the `devToken` method first, since it's a public API?
Although I do understand it was designed for Dev purposes only.

**Comment by JcMinarro** [comment]
> Hey @JcMinarro,
> 
> Shouldn't we go with the process of deprecating the `devToken` method first, since it's a public API? Although I do understand it was designed for Dev purposes only.

About the design purpose, you can read about it [here](https://getstream.io/chat/docs/android/tokens_and_authentication/?language=kotlin&q=dev#developer-tokens). Basically it is useful for devs when they are integrating our SDK with a development environment and don't want to take care of "real user tokens"

**Review by kanat — APPROVED** [review]
LGTM

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/utils/TokenUtils.kt [hunk 2]
```diff
@@ -42,4 +42,18 @@ internal object TokenUtils {
...
+     * Generate a developer token that can be used to connect users while the app is using a development environment.
+     *
+     * @param userId the desired id of the user to be connected.
+     */
+    fun devToken(userId: String): String {
+        require(userId.isNotEmpty()) { "User id must not be empty" }
+        val header = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9" //  {"alg": "HS256", "typ": "JWT"}
+        val devSignature = "devtoken"
+        val payload: String =
+            Base64.encodeToString("{\"user_id\":\"$userId\"}".toByteArray(StandardCharsets.UTF_8), Base64.NO_WRAP)
```
[the remark this answers is not in the document]
Comment (JcMinarro) [reply stream-chat-android-client/src/main/java/io/getstream/chat/android/client/utils/TokenUtils.kt:56]: It is already using `Base64.NO_WRAP`

[the remark this answers is not in the document]
Comment (amitkma) [reply stream-chat-android-client/src/main/java/io/getstream/chat/android/client/utils/TokenUtils.kt:56]: The encoded payload should be URL safe and should have no padding along with no_wrap.

**Pull request merged**